src/PostProcessors/MCSimporter.py

```diff
@@ -147,20 +148,30 @@ def mcsReader(mcsListFile):
   mcsIDs = np.zeros((0))
 
   # construct the list of MCSs and the list of BE
-  with open(mcsListFile.getFilename(), 'r') as file:
+  with open(mcsListFile, 'r') as file:
     next(file) # skip header
     lines = file.read().splitlines()
-    for l in lines:
-      elementsList = l.split(',')
-
-      mcsIDs = np.append(mcsIDs,elementsList[0])
-      elementsList.pop(0)
-
-      probability=np.append(probability,elementsList[0])
-      elementsList.pop(0)
-
-      mcsList.append(list(element.rstrip('\n') for element in elementsList))
-
-      beList.update(elementsList)
+    if type is None:
+      for l in lines:
+        elementsList = l.split(',')
+        mcsIDs = np.append(mcsIDs,elementsList[0])
+        elementsList.pop(0)
+        probability=np.append(probability,float(elementsList[0]))
+        elementsList.pop(0)
+        mcsList.append(list(element.rstrip('\n') for element in elementsList))
+        beList.update(elementsList)
+    elif type.lower() == 'saphire':
+      lines = lines[1:] # skip additional description line
+      for l in lines:
+        elementsList = l.split(',')
+        # skip empty line
+        if elementsList[0].strip() == '':
+          continue
+        mcsIDs = np.append(mcsIDs,elementsList[0])
+        probability=np.append(probability, float(elementsList[1]))
+        # skip column 3 which is the fraction to the total
+        mcs = list(element.strip() for element in elementsList[3:])
+        mcsList.append(mcs)
+        beList.update(mcs)
```

Comment:
is this line correct? mcsList and beList are the same at the end

Reply:
I think there are different. Here, mcsList is a list of list minimal cut sets, while beList is just a set of basic events. beList was previous defined as beList = set(). The name is confusing, but it was used in the original code.